_pages/clients/r.md

```diff
@@ -56,207 +54,219 @@ If you are running [Algorithmia Enterprise](/enterprise), you can specify the AP
+### Request Options
+
+The client exposes options that can configure algorithm requests. This includes support for changing the timeout or indicating that the API should include `stdout` in the response. In the following example, we set the timeout to 60 seconds and disabling `stdout` in the response:
 
 {% highlight r %}
-text_file <- "data://YOUR_USERNAME/nlp_directory/jack_london.txt"
+algo$setOptions(timeout=60, stdout=FALSE)
 {% endhighlight %}
 
-Next upload your local file to the data collection using the `.putFile()` method:
+You can find more details in [API Docs](/developers/api/?python) > [Invoke an Algorithm](/developers/api/?python#invoke-an-algorithm).
+
+### Error Handling
+
+To be able to better develop across languages, Algorithmia has created a set of standardized errors that can be returned by either the platform or by the algorithm being run. We can then catch these errors as follows:
 
 {% highlight r %}
-# Upload local file
-client$file(text_file)$putFile("/your_local_path_to_file/jack_london.txt")
+tryCatch({
+    # Run your code
+    client$algo('util/whoopsWrongAlgo')$pipe('Hello, world!')
+  }, error = function(e) {
+    stop(e))
+})
+
+Error in getResponse(client$postJsonHelper(algoUrl, input, queryParameters)) : 
+  Algorithm error: algorithm algo://util/whoopsWrongAlgo/ not found
 {% endhighlight %}
 
-This endpoint will replace a file if it already exists. If you wish to avoid replacing a file, check if the file exists before using this endpoint.
-{: .notice-warning}
+You can read more about error handling in the [Developer Center](https://algorithmia.com/developers) under [Error Handling](/developers/algorithm-development/algorithm-errors) and in the [R Algorithm Development](https://algorithmia.com/developers/algorithm-development/languages/r#error-handling) guide.
 
-You can confirm that the file was created by navigating to Algorithmia's [Hosted Data Source](/data/hosted) and finding your data collection and file.
+### Limits
 
-You can also upload your data through the UI on Algorithmia's [Hosted Data Source](/data/hosted). For instructions on how to do this go to the [Hosted Data Guide]({{site.baseurl}}/data/hosted).
+Your account can make up to {{site.data.stats.platform.max_num_algo_requests}} Algorithmia requests at the same time (this limit <a onclick="Intercom('show')">can be raised</a> if needed).
 
-### Downloading Data from a Data Collection
+## Working with Algorithmia Data Sources
 
-Next check if the file that you just uploaded to data collections exists, and try downloading it to a (new) local file:
+For some algorithms, passing input to the algorithm at request time is sufficient, while others might have larger data requirements or need to preserve state between calls. Application developers can use Algorithmia's [Hosted Data](/developers/data/hosted) to store data as text, JSON, or binary, and access it via the Algorithmia [Data API](/developers/api/?python#data).
```

Comment:
again, lets link to curl instead of python (if they can better translate the python example they can always click that tab https://algorithmia.com/developers/api/?shell#data